roles/lib_openshift_api/library/oc_secrets.py

```diff
@@ -0,0 +1,360 @@
+#
+#  # To get a specific secret named 'mysecret'
+#  - name: list secrets
+#    oca_secrets:
+#      state: list
+#      namespace: default
+#      name: mysecret
+#
+#   # To create a secret:
+#   # This module expects the user to place the files on the remote server and pass them in.
+#  - name: create a secret from file
+#    oca_secrets:
+#      state: present
+#      namespace: default
+#      name: mysecret
+#      files:
+#      - /tmp/config.yml
+#      - /tmp/passwords.yml
+#      delete_after: False
+
+#   # To create a secret:
+#   # This module expects the user to place the files on the remote server and pass them in.
+#  - name: create a secret from content
+#    oca_secrets:
+#      state: present
+#      namespace: default
+#      name: mysecret
+#      contents:
+#      - path: /tmp/config.yml
+#        content: "value=True\n"
+#      - path: /tmp/passwords.yml
+#        content: "test1\ntest2\ntest3\ntest4\n"
+#
+
+import os
+import shutil
+import json
+import atexit
+
+class OpenShiftOC(object):
+    ''' Class to wrap the oc command line tools
+    '''
+    def __init__(self,
+                 namespace,
+                 secret_name=None,
+                 kubeconfig='/etc/origin/master/admin.kubeconfig',
+                 verbose=False):
+        ''' Constructor for OpenshiftOC '''
+        self.namespace = namespace
+        self.name = secret_name
+        self.verbose = verbose
+        self.kubeconfig = kubeconfig
+
+    def get_secrets(self):
+        '''return a secret by name '''
+        cmd = ['get', 'secrets', '-o', 'json', '-n', self.namespace]
+        if self.name:
+            cmd.append(self.name)
+
+        rval = self.oc_cmd(cmd, output=True)
+
+        # Ensure results are retuned in an array
+        if rval.has_key('items'):
+            rval['results'] = rval['items']
+        elif not isinstance(rval['results'], list):
+            rval['results'] = [rval['results']]
+
+        return rval
+
+    def delete_secret(self):
+        '''return all pods '''
+        return self.oc_cmd(['delete', 'secrets', self.name, '-n', self.namespace])
+
+    def secret_new(self, files):
+        '''Create a secret with  all pods '''
+        secrets = ["%s=%s" % (os.path.basename(sfile), sfile) for sfile in files]
+        cmd = ['-n%s' % self.namespace, 'secrets', 'new', self.name]
+        cmd.extend(secrets)
+
+        return self.oc_cmd(cmd)
+
+    @staticmethod
+    def create_files_from_contents(data):
+        '''Turn an array of dict: filename, content into a files array'''
+        files = []
+        for sfile in data:
+            with open(sfile['path'], 'w') as fds:
+                fds.write(sfile['content'])
+            files.append(sfile['path'])
+
+        # Register cleanup when module is done
+        atexit.register(OpenShiftOC.cleanup, files)
+        return files
+
+    def update_secret(self, files):
+        '''run update secret '''
+        result = self.delete_secret()
```

Comment:
update_secret is extremely clean in this method.  call delete and call secret_new.  2 code paths.  By introducing the replace command this is what it would look like:
-----
        secret = self.prep_secret(files)
        if secret['returncode'] != 0:
            return secret

        sfile_path = '/tmp/%s' % secret['results']['metadata']['name']
        with open(sfile_path, 'w') as sfd:
            sfd.write(json.dumps(secret['results']))

        cmd = ['replace', 'secrets', '-f', sfile_path]

        atexit.register(OpenShiftOC.cleanup, [sfile_path])
        return self.oc_cmd(cmd)
-----